padertorch/data/segment.py

```diff
@@ -392,9 +424,75 @@ def get_segment_boundaries(
     return boundaries
 
 
+def _get_segment_length_for_mode(num_samples, length: int, shift: int = None,
+                                 length_mode: str = 'constant'):
+    """
+    This function calculates an optimal segment length assuming that `length`
+    is equal to the segment shift. Length can be used in three different ways
+    depending on the length_mode.
+
+    Args:
+        num_samples: number of samples to be segmented
+        length: segment length
+        shift: shift between segments
+        length_mode: constant: uses `length` for all examples
+                     max:      uses some length smaller than `length` for each
+                               example
+                     min:      uses some length larger than `length` for each
```

Comment:
Does min make sense? I'd assume that we can then just use the full length for a segment

Reply:
Or is this formulated like: maximize the number of segments with the constraint `new_length >= length`?

Reply:
yes, we look at the number of times length fits into num_samples and than extent the length by the residual divided by the number of time length fits in to num_samples